packages/pwa-buildpack/lib/WebpackTools/plugins/ServiceWorkerPlugin.js

```diff
@@ -1,34 +1,54 @@
-// TODO: (p1) write test file and test
+const path = require('path');
+const webpack = require('webpack');
 const WorkboxPlugin = require('workbox-webpack-plugin');
 const WriteFileWebpackPlugin = require('write-file-webpack-plugin');
+
+const createFileHash = require('../../Utilities/createFileHash');
 const optionsValidator = require('../../util/options-validator');
 
 // No longer modifiable since v4
-const SW_FILENAME = 'sw.js';
+let SERVICE_WORKER_FILENAME = 'sw.js';
+let SERVICE_WORKER_HASH = '';
+
+const DEFAULT_SW_FILE_PATH = path.join(
+    path.resolve(__dirname).split('/packages')[0],
+    'packages',
+    'venia-concept',
```

Comment:
What about projects that aren't `venia-concept`? Like if someone uses the new CLI to start their own project - are they intended to use this plugin too?

Reply:
If someone does that, will the folder name change to something else?

I was in an assumption that `venia-concept` will remain as its own folder because it is being used in multiple places in `pwa-buildpack`.

If that's not the case, first we will have to save the name of the project in a global variable somewhere are start using that in the places where we are using `venia-concept` right now.

Correct me if I am wrong on this.

Reply:
I think this is fine for now but I unresolved the conversation in case we want to continue discussing it.